rplugin/python3/deoplete/sources/deoplete_jedi/cache.py

```diff
@@ -169,7 +169,7 @@ def cache_processor_thread(compl_queue):
             last_clear = time.time()
 
         try:
-            compl = compl_queue.get(timeout=0.01)
+            compl = compl_queue.get()
```

Comment:
This can't simply be changed to blocking.  I think it should be moved to its own thread where it can block.

Reaping needs to happen mainly for scoped completions since their cache keys are based on the scope's line numbers at the time of creation.  It's not (reasonably) possible to replace existing entries for the same scope so there will be cache entries that are essentially useless that will accumulate over time.

The reaping thread could use `time.sleep()` which should further reduce the CPU usage.  I think every 30 seconds might be a little aggressive and it could be more like every 5 minutes instead.